src/main/scala/analysis/MemoryRegionAnalysis.scala

```diff
@@ -195,26 +235,93 @@ trait MemoryRegionAnalysis(val program: Program,
 //          }
           if (directCall.target.name == "malloc") {
             evaluateExpression(mallocVariable, constantProp(n)) match {
-              case Some(b: BitVecLiteral) => regionLattice.lub(s, Set(HeapRegion(nextMallocCount(), b, IRWalk.procedure(n))))
+              case Some(b: BitVecLiteral) =>
+                val negB = if isNegative(b) then b.value - BigInt(2).pow(b.size) else b.value
+                val newHeapRegion = HeapRegion(nextMallocCount(), negB, IRWalk.procedure(n))
+                addReturnHeap(directCall, newHeapRegion)
+                regionLattice.lub(s, Set(newHeapRegion))
               case None => s
```

Comment:
The heap identification currently crashes for a lot of the tests that have malloc in them. correct/malloc_with_local2/clang:BAP is one of these.

This appears to be because of this section of code - heap regions are not created unless the size parameter of malloc can be determined by constant propagation. We need to be able to handle cases where a heap region's size is not known (at least initially). In the example correct/malloc_with_local2/clang:BAP constant propagation fails to determine the size because the size parameter is put on the stack and then loaded from the stack.

Reply:
This is still an issue

Reply:
Fixed in the coming push